### PR TITLE
Fix vi-mode's word-related motions

### PR DESCRIPTION
**Pull request by fukamachi**
Fix vi-mode's word-related motions, like `e`, `E`, `w`, `W`, `b` and `B`.

The characters consist of `word` is determined by the `iskeyword` option, added by #966.

### CODE REVIEW

File: extensions/vi-mode/word.lisp [hunk 1]
```diff
@@ -1,57 +1,75 @@
...
+      (t :non-word))))
+
+(defun broad-char-type (char)
+  (when char
+    (cond
+      ((blank-char-p char) :blank)
+      (t :word))))
+
+(defun forward-word-begin (char-type-fn &optional (point (current-point)))
+  (symbol-macrolet ((point-char (character-at point)))
```
Comment (cxxxr) [inline extensions/vi-mode/word.lisp:40]: This is a very cool way to write, but I am concerned that others will see this and copy it elsewhere.
What about using labels/flet?

**Review by cxxxr — COMMENTED** [review]
Thanks for the great PR!
It's really good that they added the test.

File: extensions/vi-mode/word.lisp (hunk 1 above)
Comment (fukamachi) [reply extensions/vi-mode/word.lisp]: Okay, I replaced them with simple `flet`s.
Actually, they didn't need symbol macros, I don't know I wrote them with it.

**Comment by cxxxr** [comment]
Thank you so much!

**Pull request merged**